public/app/features/annotations/annotations_srv.ts

```diff
@@ -66,6 +84,27 @@ export class AnnotationsSrv {
       });
   }
 
+  /** if event has panel id and query is of type dashboard then panel and requesting panel id must match */
+  private matchPanelId(annotation: AnnotationEvent, panelId: number) {
+    return !annotation.panelId || annotation.source.type !== 'dashboard' || annotation.panelId === panelId;
+  }
+
+  private matchPanelAnnotationTags(
+    annotation: AnnotationEvent,
+    filterTags: string[],
+    matchAny: boolean,
+    scopedVars: ScopedVars | undefined
+  ) {
+    if (!_.isArray(filterTags) || filterTags.length === 0) {
```

Comment:
Try to limit the use of lodash when there is no real need.
```suggestion
    if (!Array.isArray(filterTags) || filterTags.length === 0) {
```